doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,223 @@
+So the correct changeset should look like:
+
+| Added Keys | Removed Keys | Modified Keys |
+| ---------- | ------------ | ------------- |
+|            |              |               |
+
+### Reproducible Example
+
+The following example will demonstrate the problematic sequence as real, runnable code.
+The only modification to the 'pseudo'-example above is that we modify something in keyset `a`, otherwise the optimizations within Elektra would never execute the whole of `kdbSet`.
+
+First, run some setup steps:
+
+```sh
+# enable the dbus plugin as a notification-send hook
+kdb set system:/elektra/hook/notification/send/plugins/#0 dbus
+
+# mount user:/a and user:/b as two backends
+kdb mount a.ecf user:/a
+kdb mount b.ecf user:/b
+
+# populate user:/a and user:/b with values
+kdb set user:/a/brightness 1
+kdb set user:/a/saturation 1
+kdb set user:/b/foreground 1
+kdb set user:/b/background 1
+```
+
+Then, run the following command to monitor the notifications by the `dbus` plugins:
+
+```sh
+dbus-monitor --session type='signal',interface='org.libelektra'
+```
+
+Finally, execute the following program:
+
+```c
+#include <kdb.h>
+#include <stdio.h>
+
+int main (void)
+{
+	Key * root = keyNew ("/", KEY_END);
+	KDB * handle = kdbOpen (NULL, root);
+
+	Key * keyA = keyNew ("user:/a", KEY_END);
+	Key * keyB = keyNew ("user:/b", KEY_END);
+
+	KeySet * a = ksNew (0, KS_END);
+	KeySet * b = ksNew (0, KS_END);
+
+	kdbGet (handle, a, keyA);
+	kdbGet (handle, b, keyB);
+
+	// we need to change *something*, otherwise the backend would assume nothing changed
+	// and never even execute all of kdbSet
+	ksAppendKey (a, keyNew ("user:/a/test", KEY_VALUE, "hi", KEY_END));
+
+	kdbSet (handle, a, keyA);
+
+	ksDel (a);
+	ksDel (b);
+	keyDel (keyA);
+	keyDel (keyB);
+	keyDel (root);
+
+	kdbClose (handle, 0);
+	return 0;
+}
+```
+
+You will notice the following output of `dbus-monitor`:
+
+```
+signal time=1666993066.749997 sender=:1.364 -> destination=(null destination) serial=2 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyAdded
+   string "user:/a/brightness"
+signal time=1666993066.750161 sender=:1.364 -> destination=(null destination) serial=3 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyAdded
+   string "user:/a/saturation"
+signal time=1666993066.750551 sender=:1.364 -> destination=(null destination) serial=4 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyAdded
+   string "user:/a/test"
+signal time=1666993066.750628 sender=:1.364 -> destination=(null destination) serial=5 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyDeleted
+   string "user:/b/background"
+signal time=1666993066.750866 sender=:1.364 -> destination=(null destination) serial=6 path=/org/libelektra/configuration; interface=org.libelektra; member=KeyDeleted
+   string "user:/b/foreground"
```

Comment:
Since we also know **why** `dbus` behaves like this (and in fact knew that before the example), we can already say that it affects all plugins that store the `KeySet` during `kdbGet` and compare against `kdbSet`. The exact behaviour will depend on the plugin.

Reply:
The problem is not that either of us doesn't know it. The problem is that it is not properly written in the decision.

Reply:
> The problem is not that either of us doesn't know it.

Your question was phrased in a way that I wasn't sure, whether you know the answer or not. 

> The problem is that it is not properly written in the decision.

IMO the general section ("Assume KDB contains the following keys:" etc.) shows pretty well that this is a general problem.

But as I said in another comment, maybe swapping the other (concrete example first, general issue afterwards) would make it more clear.

Reply:
I show that the same behavior is also present in `logchange` in 7784ea82